src/migrations/20230306103400-add-project-column-to-segments.js

```diff
@@ -0,0 +1,11 @@
+exports.up = function (db, cb) {
+    db.runSql(
+        `ALTER TABLE segments ADD COLUMN IF NOT EXISTS segment_project_id VARCHAR(255);`,
```

Comment:
USE text as type here. Also, don't use if not exists, if the column exist I believe we want the migration to fail. Also, I believe we should put an index on this field

Reply:
I've chosen `VARCHAR(255)` to use the same type as projects table: https://github.com/Unleash/unleash/blob/e4b84bc2a3620336b36067afbfd11cf5d2edd47f/src/migrations/20200928194947-add-projects.js#L10-L16

And also following other references to project.id https://github.com/Unleash/unleash/blob/e4b84bc2a3620336b36067afbfd11cf5d2edd47f/src/migrations/20210618091331-project-environments-table.js#L5 but in trying to fix the test issues I've accidentally removed the cascading configuration. 

Adding an index is worthwhile

Reply:
I looked at the different access patterns we have from the issues in our backlog, and I believe the main one is "get all segments except the ones bound to other projects", which can be solved by a single query or two queries: one by project id and another one for `project_id is NULL`. Only in the last case, the index will be helpful. So I'd like to delay the index creation until we tackle https://linear.app/unleash/issue/2-737/when-listing-segments-project-specific-segments-should-also-be-present

I'm still open to add the index but perhaps the FK constraint is enough